app/javascript/shared/helpers.ts

```diff
@@ -9,3 +9,10 @@ export function assert<T>(value: T | undefined | null): T {
 
   return value;
 }
+
+export function typesafeAssign<T extends object>(
```

Comment:
TypeScript doesn't strictly enforce types for `Object.assign(...)`, i.e. it doesn't check that the objects are of the same (or similar) type. This function does.